articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,147 @@
+# Deploying Multiple Applications to VM with NGINX as a Reverse Proxy
+
+## Introduction
+Ever wondered how more than one application is being deployed to the same machine, and how traffic is being routed to the corresponding applications?
+Keep reading to know all about it!
+## Goals
+By the end of the article, one would be able to understand
+- What a Reverse Proxy is?
+- What NGINX is?
+- How does it help in managing multiple applications?
+- How to leverage NGINX as a Reverse Proxy?
+
+### Reverse Proxy
+A reverse proxy is a type of proxy server that retrieves resources on behalf of a client from one or more servers. These resources are then returned to the client, appearing as if they originated from the server itself.   
+
+### NGINX
+We will be using NGINX which would be serving the purpose of a Reverse Proxy. NGINX is a web server that can be used as a reverse proxy, load balancer, mail proxy and HTTP cache. 
+
+### Example
+![image](./reverse_proxy_working.png)
+
+This is an example of an architecture, where two apps are running in the background, but clients have no idea about them. Clients only know about NGINX which acts as a reverse proxy that sends the request to the appropriate application.
+
+
+Now that you have a broad idea of what we are about to build, let us jump right in!
+
+## Aim
+1. Deploy two applications and have it managed by NGINX.
+
+## Setup & Pre-Requisites:
+
+1. For this example, I have two sample NodeJS servers. One can have any kind of application running on different ports. 
```

Comment:
Link to a Section article about how to setup a Node.js server.

Reply:
done and thanks